Fallback to com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl

if newInstance fails in XSLTransform constructor. I saw this happen:

```
BUILD SUCCESSFUL
Total time: 35 seconds
~/xom$ more website/build/error.txt
nu.xom.xslt.XSLException: Could not locate a TrAX TransformerFactory
nu.xom.xslt.XSLException: Could not locate a TrAX TransformerFactory
        at nu.xom.xslt.XSLTransform.<init>(Unknown Source)
        at nu.xom.xslt.XSLTransform.<init>(Unknown Source)
        at nu.xom.xslt.XSLTransform.<init>(Unknown Source)
        at nu.xom.samples.Transformer.main(Unknown Source)
Caused by: javax.xml.transform.TransformerFactoryConfigurationError: Provider org.apache.xalan.processor.TransformerFactoryImpl not found
        at java.xml/javax.xml.transform.FactoryFinder.newInstance(FactoryFinder.java:177)
        at java.xml/javax.xml.transform.FactoryFinder.find(FactoryFinder.java:212)
        at java.xml/javax.xml.transform.TransformerFactory.newInstance(TransformerFactory.java:126)
        ... 4 more
Caused by: java.lang.ClassNotFoundException: org/apache/xalan/processor/TransformerFactoryImpl
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at java.xml/javax.xml.transform.FactoryFinder.getProviderClass(FactoryFinder.java:122)
        at java.xml/javax.xml.transform.FactoryFinder.newInstance(FactoryFinder.java:165)
        ... 6 more
```

--- a/src/nu/xom/xslt/XSLTransform.java
+++ b/src/nu/xom/xslt/XSLTransform.java
@@ -176,27 +176,28 @@ public final class XSLTransform {
             factory = TransformerFactory.newInstance();
         }
         catch (TransformerFactoryConfigurationError error) {
-            try {  // fallback to system default XSLT 1.0 transformer
-                factory = TransformerFactory.newInstance(
-                  "com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl",
-                  ClassLoader.getSystemClassLoader()
+            try { // fallback to system default XSLT 1.0 transformer
+                System.setProperty(
+                  "javax.xml.transform.TransformerFactory",
+                  "com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl"
                 );
+                factory = TransformerFactory.newInstance();
             }
-            catch (TransformerFactoryConfigurationError error2) {
+            catch (TransformerFactoryConfigurationError ignored) {
                 throw new XSLException(
-                        "Could not load TransformerFactory", error2
+                        "Could not locate a TrAX TransformerFactory", error
                       ); 
             }
         } 
-        
+
         factory.setErrorListener(errorsAreFatal);
         try {
-          this.templates = factory.newTemplates(source);
+            this.templates = factory.newTemplates(source);
         }      
         catch (TransformerConfigurationException ex) {
-           throw new XSLException(
-             "Syntax error in stylesheet", ex
-           );    
+            throw new XSLException(
+              "Syntax error in stylesheet", ex
+            );    
         }
         
     }
@@ -298,7 +299,6 @@ public final class XSLTransform {
      * @param value the value of the parameter
      */
     public void setParameter(String name, String namespace, Object value) {
-       
        
        if (namespace == null || "".equals(namespace)) {
            _setParameter(name, value);

